daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/GrpcLedgerClient.scala

```diff
@@ -97,7 +97,12 @@ class GrpcLedgerClient(val grpcClient: LedgerClient, val applicationId: Applicat
                 err => throw new ConverterException(err),
                 identity,
               )
-          (ScriptLedgerClient.ActiveContract(templateId, cid, argument), key)
+          val meta =
+            NoLoggingValueValidator.validateContractMetadata(createdEvent.getMetadata) match {
```

Comment:
what happens when we run against a ledger without contract metadata?

Reply:
@rautenrieth-da what does the ledger API do for contracts created before explicit disclosure? Do you omit metadata completely? Or do you omit only the driver metadata?

Reply:
Intuitively I was expecting that they’re different. But thinking about it, I’m not sure it’s worth spending significant effort on this:

For CN we really don’t care, we’ll only have a running ledger after this. For Canton ledgers in general, we have non-empty metadata so users can distinguish between the two regardless of the encoding.

For Sandbox on X we also don’t really care since it’s a testing ledger so we care very little about persistence.

Which mainly leaves vmbc: Users wouldn’t be able to distinguish whether a contract can be used in explicit disclosure or not. That’s not amazing but also not terrible I’d say. You can still distinguish based on the offset, if you really need to.

So I’d say there is no requirement to make the two different from my side and I’d suggest to go for the simpler solution. @meiersi-da do you agree or do you think it’s worth the effort to distinguish this in the participant?

Reply:
@cocreature @rautenrieth-da : I agree that the default of an empty bytestring seems like a good choice.

Can you remind me what metadata VMBC will require? My hunch is that there won't be any metadata, and thus all contracts can be used with explicit disclosure.